pkg/controllers/awsloadbalancercontroller/deployment.go

```diff
@@ -24,6 +24,9 @@ const (
 	appLabelName    = "app.kubernetes.io/name"
 	appName         = "aws-load-balancer-operator"
 	appInstanceName = "app.kubernetes.io/instance"
+	// awsSDKLoadConfigName is the name of the environment variable which enables shared configs.
+	// Without which certain fields in the config aren't set. Eg: role_arn.
+	awsSDKLoadConfigName = "AWS_SDK_LOAD_CONFIG"
```

Comment:
Did you verify that this is actually required?

Reply:
Yes, https://github.com/aws/aws-sdk-go/blob/v1.43.42/aws/session/shared_config.go#L332 doesn't bind `role_arn` without setting `exOpts` and `AWS_SDK_LOAD_CONFIG` sets that value.